soroban-env-host/src/host.rs

```diff
@@ -238,6 +260,12 @@ impl Host {
             .map_err(Host)
     }
 
+    pub fn recover_events(self) -> Result<Events, Self> {
```

Comment:
💡 What's the significance of the term `recover` here? Is the idea that this takes the events out of the host that have been queued? I normally see the term recover used in situations where failure has occurred. Is this for a failure situation?

The term `collect` may carry less other meanings than recover.

Reply:
Yeah currently the way it's being used (e.g. in [core](https://github.com/stellar/stellar-core/blob/master/src/rust/src/contract.rs#L184)) fits `collect` or `retrieve` better. This name follows the same as `recover_storage` above. Maybe @graydon has a better perspective on how these two functions are intended to be used (e.g. if there is a failure recover part involved).

Reply:
replaced with `try_finish`. The idea here is to have a function that deconstructs a host and returns ownership of its contained objects, efficiently (without cloning them). It's a fairly normal pattern in Rust code when you're done with a thing, and we use it to convey information back to the contract-invoker in core.